twittermap/app/controllers/TwitterMapApplication.scala

```diff
@@ -25,10 +25,16 @@ class TwitterMapApplication @Inject()(val wsClient: WSClient,
   val sentimentUDF: String = config.getString("sentimentUDF").getOrElse("twitter.`snlp#getSentimentScore`(text)")
   val cities: List[JsValue] = TwitterMapApplication.loadCity(environment.getFile(USCityDataPath))
 
+  val clientLogger = Logger("client")
+
   val register = Migration_20170428.migration.up(wsClient, cloudberryRegisterURL)
   Await.result(register, 1 minutes)
 
-  def index = Action {
+  def index = Action { request =>
+    val remoteAddress = request.remoteAddress
+    val userAgent = request.headers.get("user-agent").getOrElse("unknown")
+    clientLogger.info("user IP address: " + remoteAddress)
+    clientLogger.info("user agent: " + userAgent)
```

Comment:
Can we merge these two lines into one? 

```
clientLogger.info("IP address: " + remoteAddress + "; Agent:" + userAgent)
```
Then we don't have to match two lines in case two clients visit at the same time.